### PR TITLE
Exclude lock files and don't just write changes

### PR DESCRIPTION
**Pull request by paddyroddy**
It is recommended to run the `-w` flag with also the `-i` flag https://github.com/codespell-project/codespell/blob/cb4dbe1c34911772d81b748db4c011e96ec3b54b/README.rst?plain=1#L58. I don't know how this would play with `pre-commit`, I'm tempted to remove these flags fully.

Reason being is that in #83 that this got changed https://github.com/UCL-ARC/python-tooling/pull/200/files#r1371562344 to something that it shouldn't. I've checked, and it is definitely `rouge` https://github.com/rouge-ruby/rouge.

I've also added `codespell` into the cookied repo - to keep consistent with other `pre-commit` hooks.

I've also updated the `Gemfile.lock` file - and excluded that from `codespell`.

I've also changed the YAMLs to a new way I prefer

### CODE REVIEW

**Comment by samcunliffe** [comment]
[I think it should be dropped from pre-commit entirely.](https://github.com/UCL-ARC/python-tooling/issues/54#issuecomment-1594390696)

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -6,15 +6,17 @@ repos:
...
       - id: ruff-format
-        args: ["--config", "pyproject.toml"]
+        args:
+          - --config=pyproject.toml
   - repo: https://github.com/codespell-project/codespell
     rev: v2.2.6
     hooks:
       - id: codespell
-        args: [--write-changes]
+        exclude: ".lock$"
```
Comment (paddyroddy) [inline .pre-commit-config.yaml:19]: @samcunliffe can you please look at this line

Comment (samcunliffe) [reply .pre-commit-config.yaml:19]: pre-commit will still cause the linting to fail in the case of a false positive.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -6,15 +6,17 @@ repos:
...
         exclude: "{{cookiecutter.project_slug}}/src/{{cookiecutter.package_name}}"
       - id: ruff-format
-        args: ["--config", "pyproject.toml"]
+        args:
+          - --config=pyproject.toml
   - repo: https://github.com/codespell-project/codespell
     rev: v2.2.6
     hooks:
       - id: codespell
-        args: [--write-changes]
```
Comment (paddyroddy) [inline .pre-commit-config.yaml:17]: @samcunliffe they key is to just not have this line. I've used it in `UCL-MIRSG` repos recently to great effect.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (paddyroddy) [reply .pre-commit-config.yaml:19]: Yes, but we can exclude that file in that case.

Comment (samcunliffe) [reply .pre-commit-config.yaml:19]: So we just continually add exceptions? Maybe codespell is loads better now. It's certainly actively maintained.

I still vote that it should be dropped entirely from pre-commit and just advertised as a recommended tool.

Comment (paddyroddy) [reply .pre-commit-config.yaml:19]: I do think you are talking a very niche case based on prior bad experience.

For the case of this repo (not the cookie generated one), I think it's only the `Gemfile.lock` that is mission-critical.

As I said I've added to the MIRSG general hooks, where we just exclude certificate files for everything https://github.com/UCL-MIRSG/.github/blob/35ec8a3074aa34903640bba9b78753ad43159b1e/precommit/mirsg-hooks.yaml#L1-L42

Comment (samcunliffe) [reply .pre-commit-config.yaml:19]: I can handle being overruled if that's the majority decision.

**Review by dstansby — CHANGES_REQUESTED** [review]
I think there's a few orthogonal changes, each of which deseerve their own PR so we don't get lost and to make them easier to review:

- Reformatting the `yaml`
- Removing the `--write-changes` option and excluding the lock file
- Removing `numpy` and `pydantic` from the mypy hook
- Adding `codespell` to the package template (as opposed to just using it to lint this repository)

**Comment by paddyroddy** [comment]
> I think there's a few orthogonal changes, each of which deseerve their own PR so we don't get lost and to make them easier to review:

Overkill but sure

File: .pre-commit-config.yaml (hunk 1 above)
Comment (paddyroddy) [reply .pre-commit-config.yaml:19]: Turns out this was already being done in https://github.com/UCL-ARC/python-tooling/blob/d96dd24fbd8dfe719757b3e19dae79a36ae9086a/pyproject.toml#L1-L2 and I'm assuming the PR in question merged off `main` before that entered

**Comment by paddyroddy** [comment]
Closing in favour of #210, #211, #212, #213. Have given up updating the `Gemfile.lock` as that is enough PRs as it is.

**Pull request closed without merging**